runtime/calamari/src/xcm_config.rs

```diff
@@ -208,16 +210,19 @@ impl Config for XcmExecutorConfig {
     // Defines how to Withdraw and Deposit instruction work
     type AssetTransactor = MultiAssetTransactor;
     type OriginConverter = XcmOriginToCallOrigin;
-    // Combinations of (Location, Asset) pairs which we trust as reserves.
-    type IsReserve = MultiNativeAsset;
+    // Filter to the reserve withdraw operations.
+    // This only applies to withdraw operations of foreign assets because
+    // pallet-xcm calls are filtered and xtokens does not allow sending xcm
+    // to the local chain, so we only need to support absolute views.
+    type IsReserve = MultiNativeAsset<AbsoluteReserveProvider>;
```

Comment:
Two points here:

1. This configuration hasn't really changed. The old `MultiNativeAsset` struct was just implementing the exact same code that `MultiNativeAsset<AbsoluteReserveProvider>` is using, but in `primitives/xcm.rs` . So it's a personal preference to use the upstream code, that I can revert.

2. As stated in the description the relative view is not needed for this filter, because it is only used in `ReserveAssetDeposited`, the instruction that is supposed to signify that some reserve chain has reserved some foreign assets. This can only become an issue if someone manages to send a xcm message to the origin chain, which is prevented in a bunch of ways like not having hrmp channels with yourself.